### PR TITLE
cgreen: init at 1.4.0

### PR DESCRIPTION
**Pull request by nichtsfrei**
Cgreen is a modern unit test and mocking framework for C and C++.

To make it easy to write C in a test driven fashion with proper output
and relatively easy integration Cgreen is usually a good choice. To make
it easy to write C programs on nixos it would be great to have this
library available.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -7143,6 +7143,12 @@
     githubId = 17120571;
     name = "Xinhao Luo";
   };
+  nichtsfrei = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:7146]: Please put this into a separate commit with the message ``maintainers: add nichtsfrei`` and sort it alphabetically.

Comment (nichtsfrei) [reply maintainers/maintainer-list.nix]: Thank you.
